### PR TITLE
Adapted Blueprint data pin logic to Flow

### PR DESCRIPTION
**Pull request by LindyHopperGT**
Groundwork for adding data pins to Flow.

This is only the editor-side schema code controlling how the pins connect to eachother.  Getting the data through the wires will be added in a later changelist.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,5 @@
 *.patch_*
 *.pdb
 *.target
+Source/Flow/Public/Interfaces/FlowNativeExecutableInterface.h
```
Comment (LindyHopperGT) [inline .gitignore:15]: We have these files still existing on our version, but the classes are empty (without them some of our assets won't build).

You can skip this file.

File: Source/FlowEditor/Private/Graph/FlowGraphPinFactory.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+#include "NodeFactory.h"
+#include "SGraphPin.h"
+
+#include UE_INLINE_GENERATED_CPP_BY_NAME(FlowGraphPinFactory)
+
+//////////////////////////////////////////////////////////////////////////
+// FFlowGraphPinFactory
+
+TSharedPtr<SGraphPin> FFlowGraphPinFactory::CreatePin(UEdGraphPin* InPin) const
```
Comment (LindyHopperGT) [inline Source/FlowEditor/Private/Graph/FlowGraphPinFactory.cpp:19]: I moved this code from SFlowGraphNode into a PinFactory

File: Source/FlowEditor/Public/Graph/FlowGraphSchema.h [hunk 3]
```diff
@@ -52,14 +62,45 @@ class FLOWEDITOR_API UFlowGraphSchema : public UEdGraphSchema
...
 	virtual void ForceVisualizationCacheClear() const override;
+	virtual bool ArePinsCompatible(const UEdGraphPin* PinA, const UEdGraphPin* PinB, const UClass* CallingContext = nullptr, bool bIgnoreArray = false) const override;
+	virtual void ConstructBasicPinTooltip(const UEdGraphPin& Pin, const FText& PinDescription, FString& TooltipOut) const override;
+	virtual bool IsTitleBarPin(const UEdGraphPin& Pin) const override;
+	virtual bool CanShowDataTooltipForPin(const UEdGraphPin& Pin) const override;
 	// --
 
 	// FlowGraphSchema
+
+	/**
```
Comment (LindyHopperGT) [inline Source/FlowEditor/Public/Graph/FlowGraphSchema.h:73]: These comments were from the original signatures on UEdGraphSchema_K2.  I wasn't sure if we wanted to reformat them or keep them as-is.

They aren't really "Flow Style".

**Comment by MothDoctor** [comment]
Change superseded by https://github.com/MothCocoon/FlowGraph/pull/231

**Pull request closed without merging**